src/components/carousel.tsx

```diff
@@ -4,24 +4,64 @@ import { CarouselButton } from "./carouselButton";
 import arrowLeft from "../assets/arrowLeft.png";
 import arrowRight from "../assets/arrowRight.png";
 import classNames from "classnames";
+import { CarouselItem } from "./carouselItem";
 
 interface CarouselProps {
-  items: JSX.Element[];
+  options: string[];
+  icons?: string[];
   index: number;
-  value: number;
-  valueSetter: (index: number, value: number) => void;
+  value: string;
+  valueSetter: (index: number, value: string) => void;
 }
 
-export function Carousel({ items, index, value, valueSetter }: CarouselProps) {
-  const [currentItems, setCurrentItems] = useState(items);
-  const [selectedOption, setSelectedOption] = useState(1);
+export function Carousel({
+  options,
+  icons,
+  index,
+  value,
+  valueSetter,
+}: CarouselProps) {
+  const [currentItems, setCurrentItems] = useState<JSX.Element[]>([]);
```

Comment:
I would still try to refactor it a bit more to avoid keeping JSX.Elements in the state. The state should represent dynamic data of a component, and should keep only plain JavaScript values, objects, or arrays. 

Then, in the return statement of a component you use state to draw the JSX elements

Keeping in the state react components can lead to performance issues